egs/babel_multilang/s5/local/nnet3/run_tdnn_joint_babel_sp_bnf.sh

```diff
@@ -0,0 +1,241 @@
+
+for lang in `seq 0 $[$num_lang-1]`; do
+  for f in data/${lang_list[$lang]}/train/{feats.scp,text} exp/${lang_list[$lang]}/$alidir/ali.1.gz exp/${lang_list[$lang]}/$alidir/tree; do
+   [ ! -f $f ] && echo "$0: no such file $f" && exit 1; 
+if [ "$speed_perturb" == "true" ]; then
+  suffix=${suffix}_sp
+fi
+
+# if true, ivector extractor trained on pooled data from all languages.
+ivector_suffix=${ivector_suffix}_gb
+
+if $use_pitch; then aux_suffix=${aux_suffix}_pitch ; fi
+if $use_entropy;then aux_suffix=${aux_suffix}_entropy ; fi  
+dir=${dir}${suffix}
+
+# extract high resolution MFCC features for speed-perturbed data
+# and extract alignment 
+for lang in `seq 0 $[$num_lang-1]`; do
+  local/nnet3/run_common_langs.sh --stage $stage \
+    --speed-perturb $speed_perturb ${lang_list[$lang]} || exit;
+done
+
+# combine training data for all langs for training global i-vector extractor
+echo suffix = $suffix
+if [ ! -f data/multi/train${suffix}_hires/.done ]; then
+  echo ---------------------------------------------------------------------
+  echo "Pooling training data in data/multi${suffix}_hires on" `date`
+  echo ---------------------------------------------------------------------
+  mkdir -p data/multi
+  mkdir -p data/multi/train${suffix}_hires
+  combine_lang_list=""
+  for lang in `seq 0 $[$num_lang-1]`;do
+    combine_lang_list="$combine_lang_list data/${lang_list[$lang]}/train${suffix}_hires"
+  done
+  utils/combine_data.sh data/multi/train${suffix}_hires $combine_lang_list
+  utils/validate_data_dir.sh --no-feats data/multi/train${suffix}_hires
+  touch data/multi/train${suffix}_hires/.done
+fi
+# If we do not use separate initial layer per language
+# then we use ivector extractor trained on pooled data from all languages
+# using an LDA+MLLT transform arbitrarily chonsed from single language.
+echo hi
+if [ ! -f $global_extractor/.done ]; then
+  echo "Generate global i-vector extractor"
+  local/nnet3/run_shared_ivector_extractor.sh --global-extractor $global_extractor \
+    --stage $stage $lda_mllt_lang || exit 1; 
+  touch $global_extractor/.done
+fi
+
+# extract ivector for all languages.
+for lang in `seq 0 $[$num_lang-1]`; do
+  local/nnet3/run_ivector_common_langs.sh --stage $stage \
+    --global-extractor $global_extractor \
+    --speed-perturb $speed_perturb ${lang_list[$lang]} || exit;
+done
+
+# set num_leaves for all languages
+for lang in `seq 0 $[$num_lang-1]`; do
+  num_leaves=`tree-info exp/${lang_list[$lang]}/$alidir/tree 2>/dev/null | grep num-pdfs | awk '{print $2}'` || exit 1;
+  num_multiple_leaves="$num_multiple_leaves $num_leaves"
+
+  multi_egs_dirs[$lang]=exp/${lang_list[$lang]}/nnet3/egs_v${avg_num_archives}${ivector_suffix}
+  multi_ali_dirs[$lang]=exp/${lang_list[$lang]}/tri5_ali${suffix}
+
+done
+
+online_ivector_dir=
+if $use_ivector; then
+  online_ivector_dir=exp/${lang_list[0]}/nnet3/ivectors_train${suffix}${ivector_suffix}
+fi
+if [ -z "${online_ivector_dir}" ]; then
```

Comment:
Add print statements to describe what is happening. 

e.g. `$0: Not using ivectors as no --online-ivector-dir has been specified.`